zipline/finance/metrics/metric.py

```diff
@@ -448,15 +448,18 @@ def end_of_bar(self,
                    session_ix,
                    data_portal):
         risk = packet['cumulative_risk_metrics']
-        alpha, beta = ep.alpha_beta_aligned(
-            ledger.daily_returns_array[:session_ix + 1],
-            self._daily_returns_array[:session_ix + 1],
-        )
-
-        if np.isnan(alpha):
-            alpha = None
-        if np.isnan(beta):
-            beta = None
+        if np.any(self._daily_returns_array):
```

Comment:
You are right Scott. np.any() has a cost even if it stops when a [non-null value is found.](https://stackoverflow.com/questions/17128116/why-is-numpy-any-so-slow-over-large-arrays). It makes sense to not do this expensive check as most of the runs would be with benchmark returns. 

Ideally, it would have been better to know at this level if the parameter no-benchmark is used and skip the calculation. For the NaN option, would it be more efficient?

In any case, I'll remove this check for the time being and put it separately as a nice to have. The calls don't fail when the `daily_returns_array` is zero. 

ref(): AlphaBeta.end_of_bar without the modification
new(): AlphaBeta.with the modification 

**With benchmark returns** 
```
%timeit ref()
28.8 µs ± 836 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
%timeit new()
32.7 µs ± 192 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

**With no-benchmark returns**
```
%timeit ref()
29 µs ± 287 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
%timeit new()
2.9 µs ± 22.7 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```